src/cdk-experimental/menu/menu-item-trigger.ts

```diff
@@ -92,6 +92,24 @@ export class CdkMenuItemTrigger implements OnDestroy {
 
     this._overlayRef = this._overlay.create(this._getOverlayConfig());
     this._overlayRef.attach(this._getPortal());
+
+    this._configureMenu();
+    this._subscribeToClosingActions();
+  }
+
+  /** Wire up the subscription which can cause the menu to close. */
+  private _subscribeToClosingActions() {
+    if (this._overlayRef) {
+      this._overlayRef.outsidePointerEvents().subscribe(() => this.isMenuOpen() && this.toggle());
+    }
+  }
+
+  /** Set the elements array on the open menu and push the open menu to the array. */
+  private _configureMenu() {
+    if (this._menuPanel && this._menuPanel._menu) {
+      this._parentMenu._openMenuElements.push(...this._menuPanel._menu._openMenuElements);
```

Comment:
Is there a straightforward way to use methods to add menu elements to a parent's openMenuElements instead of reaching into its fields directly?

Reply:
+1, would be better to surface this through an API rather than mutating the field directly. Alternatively, it might make sense to have a small internal menu singleton service that just keeps track of the open menu panels (potentially as a layer between the trigger and `Overlay`)